index/rolodex_remote_loader.go

```diff
@@ -462,5 +459,10 @@ func (i *RemoteFS) Open(remoteURL string) (fs.File, error) {
 			i.rolodex.AddExternalIndex(idx, remoteParsedURL.String())
 		}
 	}
+
+	// remove from processing
+	processingWaiter.file = remoteFile
+	processingWaiter.done = true
+	i.ProcessingFiles.Delete(remoteParsedURL.Path)
```

Comment:
@daveshanley: is this maybe what caused the race condition with not sequentially resolving refs, you mentioned? I was thinking that before this change, a remote file would be marked as done and deleted from `ProcessingFiles` before the index was built. This seems racy to me. 🤔

Reply:
Very possibly yes. I am sure that waiter code could be more effectively implemented. It took me a while to stop things from locking permanently in there with strange conditions.